src/core/drive/head_snapshot.ts

```diff
@@ -74,47 +40,15 @@ export class HeadSnapshot extends Snapshot<HTMLHeadElement> {
   }
 
   findMetaElementByName(name: string) {
-    return Object.keys(this.detailsByOuterHTML).reduce((result, outerHTML) => {
-      const {
-        elements: [element],
-      } = this.detailsByOuterHTML[outerHTML]
-      return elementIsMetaElementWithName(element, name) ? element : result
-    }, undefined as Element | undefined)
+    return this.getElements("meta").filter((e) => (e as HTMLMetaElement).name == name)[0] as Element | undefined
```

Comment:
Thank you for suggesting these changes!

Since we're grouping tag names, have you considered using class names as keys instead of strings? `HTMLMetaEelement` instead of `"meta"`, `HTMLScriptElement` instead of `"script"`, etc.

That way, when we read the values from the arrays, we don't need to cast them.

Reply:
Since this is grabbing the first occurrence, have you considered using [Array.find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) instead of `filter` and index access?

Reply:
Yes.  Exactly what I wanted to implement, but I get a typescript error when using `element.constructor` as a key.  Would you know another method to accomplish this?

```
(!) Plugin typescript: @rollup/plugin-typescript TS2538: Type 'Function' cannot be used as an index type.
src/core/drive/head_snapshot.ts: (18:42)

18         const collection = this.elements[element.constructor] || []
```

Reply:
Yes.  Exactly what I wanted to implement.  Thanks for the pointers, however, I get this error using the `extends Element` recommendation:
```
(!) Plugin typescript: @rollup/plugin-typescript TS1005: '(' expected.
src/core/drive/head_snapshot.ts: (4:40)

4   elements: { [key: <E extends Element>]: E[] } = []
```


My original attempt was to use `function` as a key and get the type with `element.constructor` - but that fails with this error. 

```
(!) Plugin typescript: @rollup/plugin-typescript TS2538: Type 'Function' cannot be used as an index type.
src/core/drive/head_snapshot.ts: (18:42)

18         const collection = this.elements[element.constructor] || []
```